packages/electron/src/electron/main/index.ts

```diff
@@ -1,12 +1,82 @@
-import fs from "node:fs";
 import path from "node:path";
 import * as Sentry from "@sentry/electron/main";
 import { clearBuffer, setupSidecar } from "@spotlightjs/sidecar";
 import { BrowserWindow, Menu, Tray, app, dialog, ipcMain, nativeImage, shell } from "electron";
 import Store from "electron-store";
+import { autoUpdater } from "electron-updater";
 
 const store = new Store();
 
+autoUpdater.forceDevUpdateConfig = process.env.NODE_ENV === "development";
+
+const ONE_HOUR = 60 * 60 * 1000;
+
+function installAndRestart() {
+  /**
+   * On macOS 15+ auto-update / relaunch issues:
+   * - https://github.com/electron-userland/electron-builder/issues/8795
+   * - https://github.com/electron-userland/electron-builder/issues/8997
+   */
+  if (process.platform === "darwin") {
+    app.removeAllListeners("before-quit");
+    app.removeAllListeners("window-all-closed");
+
+    for (const win of BrowserWindow.getAllWindows()) {
+      if (win.isDestroyed()) continue;
+      win.removeAllListeners("close");
+      win.close();
+    }
+  }
+
+  autoUpdater.quitAndInstall();
+}
+
+async function checkForUpdates() {
+  try {
+    await autoUpdater.checkForUpdates();
+  } catch (error) {
+    console.error(error);
+    Sentry.captureException(error);
+  }
+
+  setTimeout(checkForUpdates, ONE_HOUR);
```

Comment:
We should have a global `let CHECK_UPDATES_TIMER: Timeout | null = null` and then do
```
if (CHECK_UPDATES_TIMER) {
  clearTimeout(CHECK_UPDATES_TIMER);
  CHECK_UPDATES_TIMER = null;
}
....
CHECK_UPDATES_TIMER = setTimeout(checkForUpdates, ONE_HOUR);
```

to prevent multiple concurrent check update calls getting scheduled if something else in the code tries to call `checkForUpdates()` separately.